lib/handler/mruby.c

```diff
@@ -698,6 +698,15 @@ static void send_response(h2o_mruby_generator_t *generator, mrb_int status, mrb_
 
     /* use fiber in case we need to call #each */
     if (!mrb_nil_p(body)) {
+        /* check the body hasn't already consumed */
+        mrb_sym consume = mrb_intern_lit(mrb, "consume!");
+        if (mrb_respond_to(mrb, body, consume)) {
+            mrb_funcall_argv(mrb, body, consume, 0, NULL);
```

Comment:
How about doing the check in C side instead?

I'd assume that it would be trivial to do so in C, and that from the mruby side you can instruct the C side to do the check possible by adding a boolean argument to `H2O_MRUBY_CALLBACK_ID_HTTP_FETCH_CHUNK` if the check should be performed, and by only setting the argument to true on the first call.

The reasons I suggest the approach are:
* we should optimize for fast path; however, calling a mruby method from C is slow
* it is preferable not to expose the interface we use for checking, since the name of the method used for checking might conflict with that the user defines